packages/frontend/src/utils/fiatValueManager.js

```diff
@@ -0,0 +1,118 @@
+            return cache.set(...args);
+        },
+
+        delete: (...args) => {
+            return cache.del(...args);
+        },
+
+        clear: (...args) => {
+            return cache.flushAll(...args);
+        }
+    };
+}
+
+export default class FiatValueManager {
+    constructor() {
+        this.coinGeckoFiatValueDataLoader = new DataLoader(
+            async (tokenIds) => {
+                try {
+                    const tokenFiatValues = await sendJson(
+                        'GET', 
+                        stringifyUrl({
+                            url: COINGECKO_PRICE_URL,
+                            query: {
+                                ids: tokenIds.join(','),
+                                vs_currencies: 'usd,eur,cny',
+                                include_last_updated_at: true
+                            }
+                        })
+                    );
+                    return tokenIds.map((id) => tokenFiatValues[id]);
+                } catch (error) {
+                    console.error(`Failed to fetch coingecko prices: ${error}`);
+                    // DataLoader must be constructed with a function which accepts 
+                    // Array<key> and returns Promise<Array<value>> of the same length
+                    // as the Array of keys
+                    return new Promise((resolve, reject) => {
+                        return [].fill({}, tokenIds.length);
+                    });
+                }
+            },
+            {
+                /* 0 checkperiod means we only purge values from the cache on attempting to read an expired value
+                Which allows us to avoid having to call `.close()` on the cache to allow node to exit cleanly */
+                cacheMap: wrapNodeCacheForDataloader(new Cache({ stdTTL: 30, checkperiod: 0, useClones: false }))
+            }
+        );
+        this.refFinanceDataLoader = new DataLoader(
+            async (dummyToken) => {
+                try {
+                    const refFinanceTokenFiatValues = await sendJson(
+                        'GET',
+                        REF_FINANCE_API_ENDPOINT + '/list-token-price'
+                    );
+                    return [refFinanceTokenFiatValues];
+                } catch (error) {
+                    console.error(`Failed to fetch ref-finance prices: ${error}`);
+                    return new Promise((resolve, reject) => [{}]);
+                }
+            },
+            {
+                cacheMap: wrapNodeCacheForDataloader(new Cache({ stdTTL: 30, checkperiod: 0, useClones: false }))
+            }
+        );
+    };
+
+    async getPrice(tokens = ['near', 'usn']) {
+        const byTokenName = {};
+        const prices = await this.fiatValueDataLoader.loadMany(tokens);
+        tokens.forEach((tokenName, ndx) => byTokenName[tokenName] = prices[ndx]);
+        return byTokenName;
+    };
+
+    async fetchTokenPrices(dummyToken = ['near']) {
```

Comment:
nit: I think these method names could be more descriptive e.g. `fetchRefFinanceTokenPrices` or similar. That way, when we `Promise.all` or similar, it's clear why we're calling multiple methods and awaiting multiple promises